File: testbed/synthetic-vuln-java-app/src/main/java/com/example/agent/config/HardcodedKeys.java
// Anti-pattern: D005 Java (hardcoded LLM credentials — CWE-798).
package com.example.agent.config;

import com.azure.core.credential.AzureKeyCredential;
import dev.langchain4j.model.anthropic.AnthropicChatModel;
import dev.langchain4j.model.openai.OpenAiChatModel;
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;

public class HardcodedKeys {

    public OpenAiChatModel openai() {
        return OpenAiChatModel.builder()
                .apiKey("sk-proj-AAAAAAAAAAAAAAAAAAAAAAAAAAAA")
                .modelName("gpt-4o-mini")
                .build();
    }

    public AnthropicChatModel anthropic() {
        return AnthropicChatModel.builder()
                .apiKey("sk-ant-api03-XXXXXXXXXXXXXXXXX")
                .modelName("claude-sonnet-4-5")
                .build();
    }

    public AzureKeyCredential azure() {
        return new AzureKeyCredential("az-key-LITERAL-EXAMPLE-AAAA");
    }

    public AwsBasicCredentials awsBasic() {
        return AwsBasicCredentials.create(
                "AKIAIOSFODNN7EXAMPLE",
                "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
        );
    }
}

File: testbed/synthetic-vuln-python-app/src/synthetic_vuln_python_app/hardcoded_keys.py
"""D005 — hardcoded API keys in LLM client constructors."""

import boto3
import openai
import anthropic
import cohere
from langchain_openai import ChatOpenAI


def make_clients():
    a = openai.OpenAI(api_key="sk-proj-AAAAAAAAAAAAAAAAAAAAAAAAAAAA")
    b = anthropic.Anthropic(api_key="sk-ant-api03-XXXXXXXXXXXXXXXXX")
    c = cohere.Client("co-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")
    bedrock = boto3.client(
        "bedrock-runtime",
        region_name="us-east-1",
        aws_access_key_id="AKIAIOSFODNN7EXAMPLE",
        aws_secret_access_key="wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
    )
    chat = ChatOpenAI(api_key="sk-proj-BBBBBBBBBBBBBBBBBBBBBBBBBBBB")
    return a, b, c, bedrock, chat
